### PR TITLE
BSD more friendly build.

### PR DESCRIPTION
**Pull request by devnexen**
Using native feature for proper TID report.

### CODE REVIEW

File: src/libmongoc/src/mongoc/mongoc-log.c [hunk 1]
```diff
@@ -171,6 +173,12 @@ mongoc_log_default_handler (mongoc_log_level_t log_level,
    pid = syscall (SYS_gettid);
 #elif defined(_WIN32)
    pid = (int) _getpid ();
+#elif defined(__FreeBSD__)
+   long tid;
+   thr_self (&tid);
```
[the remark this answers is not in the document]
Comment (devnexen) [reply src/libmongoc/src/mongoc/mongoc-log.c]: same semantic as Linux's sys call in fact.

[the remark this answers is not in the document]
Comment (kevinAlbs) [reply src/libmongoc/src/mongoc/mongoc-log.c]: Ah I misunderstood what this code was previously doing. We're getting the thread id if possible, and using the process id as a fallback.

File: src/libmongoc/CMakeLists.txt [hunk 2]
```diff
@@ -338,7 +338,10 @@ if (ENABLE_SHM_COUNTERS STREQUAL "ON")
          "Shared memory performance counters not supported on Mac or Windows")
    endif ()
    set (MONGOC_ENABLE_SHM_COUNTERS 1)
-   set (SHM_LIBRARIES rt)
+   find_library(RT_LIBRARY rt)
+   if (RT_LIBRARY)
+   	set (SHM_LIBRARIES rt)
+   endif ()
```
Comment (kevinAlbs) [inline src/libmongoc/CMakeLists.txt]: Just to double check, librt is not necessary for `shm_open` and `shm_unlink` (which are instead part of libc). So if we cannot find librt, that is not an issue for BSD. Is that correct?

Comment (devnexen) [reply src/libmongoc/CMakeLists.txt]: Yes

**Pull request merged**